h/services/search_index/service_factory.py

```diff
@@ -9,4 +11,10 @@ def factory(_context, request):
         es_client=request.es,
         session=request.db,
         settings=request.find_service(name="settings"),
+        queue=Queue(
+            db=request.db,
+            es=request.es,
+            batch_indexer=BatchIndexer(request.db, request.es, request),
+            limit=request.registry.settings["h.es_sync_job_limit"],
+        ),
```

Comment:
I'm not sure we make much use of the composition here. It could be private I think, but also... does it matter? Makes this slightly harder to mock as we have to prepare a pretend queue too, rather than letting mock make one for us.

Reply:
Keeps the ugly stuff (constructing the `BatchIndexer`) out of the service itself, though